automation/taskcluster/decision_task.py

```diff
@@ -172,66 +129,167 @@ def variants_to_definitions(variants, run_tests=True, lint_task=None):
+        other_tasks[taskcluster.slugId()] = craft_function()
+
+    return (build_tasks, other_tasks)
+
+
+def _get_gradle_module_name(artifact_info):
+    return ':{}'.format(artifact_info['name'])
 
 
-def create_ktlint_task():
-    return create_task(
-        name='Android Components - ktlint',
-        description='Running ktlint over all modules',
-        command='ktlint')
+def _get_release_gradle_tasks(module_name, is_snapshot):
+    gradle_tasks = ' '.join(
+        '{}:{}{}'.format(
+            module_name,
+            gradle_task,
+            '' if is_snapshot else 'Release'
+        )
+        for gradle_task in ('assemble', 'test', 'lint')
+    )
 
+    return gradle_tasks + ' uploadArchives zipMavenArtifacts'
 
-def create_compare_locales_task():
-    return create_raw_task(
-        name='Android Components - compare-locales',
-        description='Validate strings.xml with compare-locales',
-        full_command='pip install "compare-locales>=5.0.2,<6.0" && compare-locales --validate l10n.toml .')
+
+def release(artifacts_info, version, is_snapshot, is_staging):
+    version = components_version() if version is None else version
+
+    build_tasks = {}
+    wait_on_builds_tasks = {}
+    beetmover_tasks = {}
+    other_tasks = {}
+    wait_on_builds_task_id = taskcluster.slugId()
+
+    for artifact_info in artifacts_info:
+        build_task_id = taskcluster.slugId()
+        module_name = _get_gradle_module_name(artifact_info)
+        build_tasks[build_task_id] = BUILDER.craft_build_task(
+            module_name=module_name,
+            gradle_tasks=_get_release_gradle_tasks(module_name, is_snapshot),
+            subtitle='({}{})'.format(version, '-SNAPSHOT' if is_snapshot else ''),
+            run_coverage=False,
+            is_snapshot=is_snapshot,
+            artifact_info=artifact_info
+        )
+
+        beetmover_tasks[taskcluster.slugId()] = BUILDER.craft_beetmover_task(
+            build_task_id, wait_on_builds_task_id, version, artifact_info['artifact'],
+            artifact_info['name'], is_snapshot, is_staging
+        )
+
+    wait_on_builds_tasks[wait_on_builds_task_id] = BUILDER.craft_wait_on_builds_task(build_tasks.keys())
+
+    if is_snapshot:     # XXX These jobs perma-fail on release
+        for craft_function in (BUILDER.craft_detekt_task, BUILDER.craft_ktlint_task, BUILDER.craft_compare_locales_task):
+            other_tasks[taskcluster.slugId()] = craft_function()
+
+    return (build_tasks, wait_on_builds_tasks, beetmover_tasks, other_tasks)
 
 
 if __name__ == "__main__":
-    if SKIP_TASKS_TRIGGER in PR_TITLE:
-        print("Pull request title contains", SKIP_TASKS_TRIGGER)
-        print("Exit")
-        exit(0)
+    parser = argparse.ArgumentParser(description='Creates and submit a graph of tasks on Taskcluster.')
+
+    subparsers = parser.add_subparsers(dest='command')
+
+    pr_or_push_parser = subparsers.add_parser('pr-or-push')
```

Comment:
Nit: this variable isn't used, I'm not sure that you need this line

Reply:
You're right. I ran flake8 quite a few times, and I'm surprised it didn't catch this. The variable assignment is useless, but the `subparsers.add_parser('pr-or-push')` statement is needed. I kept it